controllers/post.controller.js

```diff
@@ -0,0 +1,47 @@
+import Post from '../models/post.model.js';
+import { fMsg } from "../utils/libby.js";
+
+
+
+export const createPost = async (req, res) => {
+    try {
+        const post = new Post({
+            ...req.body,
+        })
+        await post.save();
+        await post.populate('posted_by', 'userName email phone roles');
+        fMsg(res, "Post created successfully", post);
+    } catch (error) {
+        console.log(error)
+        fMsg(res, "error in creating post", error);
+    }
+};
+
+export const editPost = async (req, res) => {
+    try {
+        const post = await Post.findByIdAndUpdate(req.params.post_id, {
```

Comment:
you should clarify the possible error like, Invalid Id or smth , if the post does not exist (!post){ reply to frontend smth)